README.md

```diff
@@ -4,61 +4,38 @@ This is an example of an out-of-tree [MLIR](https://mlir.llvm.org/) dialect alon
 
 ## How to Install
 
-0. Git clone llvm repository (at commit point bb59eb8ed534da2bd03117cfde594321add4d60c)
+0. Do a recursive clone
+```sh
+git clone --recursive https://github.com/finch-tensor/Finch-mlir.git
 ```
-git clone https://github.com/llvm/llvm-project.git
-git checkout bb59eb8ed534da2bd03117cfde594321add4d60c
-cd llvm-project
-```
-
-1. Install prerequisites for mlir python bindings
 
+Or if you have already cloned this repository, run
+```sh
+git submodule init --update --recursive
 ```
-# Create python virtual environment
-# Make sure your 'python' is what you expect.
-which python
-python -m venv ~/.venv/mlirdev
-source ~/.venv/mlirdev/bin/activate
 
-python -m pip install --upgrade pip
-python -m pip install -r mlir/python/requirements.txt
-```
+This will take some time as it performa a shallow-clone of LLVM.
 
-2. Build MLIR
+1. Install prerequisites to build the dialect and the Python bindings:
 
-```
-mkdir build
-cd build
-cmake -G Ninja ../llvm \                                                                                                                                                  
-   -DLLVM_ENABLE_PROJECTS=mlir \
-   -DLLVM_BUILD_EXAMPLES=ON \
-   -DLLVM_TARGETS_TO_BUILD="Native" \
-   -DCMAKE_BUILD_TYPE=Release \
-   -DLLVM_ENABLE_ASSERTIONS=ON \
-   -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
-   -DPython3_EXECUTABLE=<YOUR PYTHON PATH (e.g., /Users/jaeyeonwon/.venv/mlirdev/bin/python)>
+1a. Use your system package manager or `conda` to install a C/C++ compiler and CCache.
 
-# Build and Check mlir (I got 88% passed)
-cmake --build . --target check-mlir
+1b. Install the build-time Python requirements:
 
+```sh
+# Generate the build requirements dynamically from `pyproject.toml` and install them.
+./scripts/install_build_reqs.sh
 ```
 
-3. Add PYTHONPATH
-```
-# Add below line to .bashrc or .zshrc
-export PYTHONPATH=$LLVM_BUILD_DIR/tools/mlir/python_packages/mlir_core:$PYTHONPATH
-```
-
-
-4. Build Finch-mlir
+2. Build the wheel or install the package:
 
+```sh
+python -m build --no-isolation
 ```
-https://github.com/finch-tensor/Finch-mlir.git
-cd Finch-mlir
-mkdir build && cd build
-LLVM_BUILD_DIR=<YOUR LLVM BUILD PATH (e.g., /Users/jaeyeonwon/llvm-project/build>
-cmake -G Ninja .. -DMLIR_DIR=$LLVM_BUILD_DIR/lib/cmake/mlir -DLLVM_EXTERNAL_LIT=$LLVM_BUILD_DIR/bin/llvm-lit
 
-# Build and Check finch-mlir
-cmake --build . --target check-finch
```

Comment:
I think we still want to be able to run dialect tests locally. Does this command work?

Reply:
After building the wheel, these tests can be run with `cmake --build build --target check-finch`. If the CMake configure step needs to be run, it's easier to just run the wheel build once more.

Reply:
I'll add this to the `README`.

Reply:
Last question: If a user changes sourcecode of of finch dialect and reruns `cmake --build build --target check-finch` does it actually pick new changes and rebuild? In the command you point to `build` directory.

For development it needs to work in an `editable` mode for dialect cpp source code.

Reply:
Editable mode doesn't make sense for C++, one always has to recompile. However, changing a single file only compiles that file and its dependencies. The `finch-test` target points to the lit/FileCheck tests in `test/`, and introducing an error in `test/looplet-complex.mlir` causes a failure in `cmake --build build --target check-finch`.

This is because the `build` directory only contains the installed files and some CMake metadata, not a copy of the source code. The metadata points back at the source code in the main directory.

The common reasons to run a full build again (most of which would also be skipped and/or cached) would be e.g. to introduce a new `*.cpp`/`*.h` file, or rework the directory structure.

Reply:
Pure-Python changes will require a rebuild though, sadly.

Reply:
By "editable" here I meant running `cmake --build build --target check-finch`, changing some c++ source file and running `cmake --build build --target check-finch` again with these changes picked. Sorry for confusion.